Content.Server/Explosion/EntitySystems/TriggerSystem.TimedCollide.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Linq;
+using Content.Server.Explosion.Components;
+using Content.Server.Explosion.EntitySystems;
+using Robust.Shared.Physics.Dynamics;
+
+namespace Content.Server.Explosion.EntitySystems;
+
+public sealed partial class TriggerSystem
+{
+    //Stores the entity as active so the event can resolve for each UID colliding with this entity.
+    private readonly List<EntityUid> Active = new();
+
+    private void InitializeTimedCollide()
+    {
+        SubscribeLocalEvent<TriggerOnTimedCollideComponent, StartCollideEvent>(OnTimerCollide);
+        SubscribeLocalEvent<TriggerOnTimedCollideComponent, EndCollideEvent>(OnTimerEndCollide);
+    }
+
+    private void OnTimerCollide(EntityUid uid, TriggerOnTimedCollideComponent component, StartCollideEvent args)
+    {
+        Active.Add(uid);
+        var otherUID = args.OtherFixture.Body.Owner;
+        component.Colliding.Add(otherUID, 0);
+    }
+
+    private void OnTimerEndCollide(EntityUid uid, TriggerOnTimedCollideComponent component, EndCollideEvent args)
+    {
+        var otherUID = args.OtherFixture.Body.Owner;
+        component.Colliding.Remove(otherUID);
+
+        if (component.Colliding.Count == 0)
+        {
+            Active.Remove(uid);
+        }
+    }
+
+    private void UpdateTimedCollide(float frameTime)
+    {
+        base.Update(frameTime);
+        foreach (var trigger in Active)
+        {
+            if (!TryComp(trigger, out TriggerOnTimedCollideComponent? component))
+                continue;
+            foreach (var (collidingEntity, collidingTimer) in component.Colliding)
+            {
+                component.Colliding[collidingEntity] += frameTime;
+                if (collidingTimer > component.Threshold)
+                {
+                    RaiseLocalEvent(trigger, new TriggerEvent(trigger, collidingEntity));
+                }
+            }
```

Comment:
Should this reset the timer or remove them or something?

Reply:
It shouldn't reset the timer, it's suppose to be on a per-entity event. So if one entity overlapped, and then a different entity overlapped. Events would trigger for them at those separate times.

Reply:
Lemme playtest around with it to see what's going on and I'll get back to you.

Reply:
Oh I see now. I just tested it out with a Flash On Trigger without a delete. I'll see what I can do.

Reply:
I think I solved it for now?

It'll reset their timer so that it won't keep them in an infinite loop.

Though I guess it will if there's a stuntimer or flashtimer that's longer or equal to the threshold.